File: dataframes/week12_simpleAgg.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.SparkConf

object week12_simpleAgg extends App {

    val sparkconf = new SparkConf
    sparkconf.set("spark.app.name" , "invoice")
    sparkconf.set("spark.master" , "local[*]")

    val spark = SparkSession.builder()
    .config(sparkconf)
    .getOrCreate()

    val invoice = spark.read.
    format("csv").
    option("path" , "c:\\Users\\pp255070\\OneDrive - Teradata\\Documents\\Spark\\data folder\\week12\\order_data-201025-223502.csv").
    option("header", true).
    option("inferSchema" , true).
    load()

    // column object method

    invoice.select(
        count("*").as("RowCount"),
        sum("Quantity").as("TotalQuantity"),
        avg("UnitPrice").as("Average"),
        countDistinct("InvoiceNo").as("DistinctInvoice")).
        show()

    // column string method

    invoice.selectExpr(
        "count(*) as RowCount",
        "sum(Quantity) as TotalQuantity",
        "avg(UnitPrice) as Average",
        "count(Distinct(InvoiceNo)) as DistinctInvoice").
        show()

    // Spark Sql , easier way as compared to dataframne and datasets

    invoice.createOrReplaceTempView("sales")

    spark.sql(" select count(*), sum(Quantity) , avg(UnitPrice) , count(Distinct(InvoiceNo)) from sales").show()
}

File: dataframes/week12_simpleAgg.py
from pyspark import SparkConf
from pyspark.sql import SparkSession
from pyspark.sql.functions import *

my_conf = SparkConf()
my_conf.set("spark.app.name" ,"my_app")
my_conf.set("spark.master" , "local[*]")

spark = SparkSession.builder.config(conf=my_conf).getOrCreate()

invoice = spark.read.\
    format("csv").\
    option("path" , "c:\\Users\\pp255070\\OneDrive - Teradata\\Documents\\Spark\\data folder\\week12\\order_data-201025-223502.csv").\
    option("header", True).\
    option("inferSchema" , True).\
    load()

  # column object method

invoice.select(
    count("*").alias("RowCount"),
    sum("Quantity").alias("TotalQuantity"),
    avg("UnitPrice").alias("Average"),
    countDistinct("InvoiceNo").alias("DistinctInvoice")).\
    show()

# column string method

invoice.selectExpr(
        "count(*) as RowCount",
        "sum(Quantity) as TotalQuantity",
        "avg(UnitPrice) as Average",
        "count(Distinct(InvoiceNo)) as DistinctInvoice").\
        show()

# Spark Sql , easier way as compared to dataframne and datasets

invoice.createOrReplaceTempView("sales")

spark.sql(" select count(*), sum(Quantity) , avg(UnitPrice) , count(Distinct(InvoiceNo)) from sales").show()